contracts/moda-contracts/src/Profile.sol

```diff
@@ -52,6 +95,18 @@ contract Profile is IProfile, IERC721, IERC721Metadata, IERC4906, ERC165 {
         emit MetadataUpdate(tokenId);
     }
 
+    /**
+     * @dev See {IProfile-updateProfileFor}.
+     */
+    function updateProfileFor(address kontract, string memory uri) external requireAuthority(kontract) {
```

Comment:
And here https://medium.com/coinmonks/solidity-storage-vs-memory-vs-calldata-8c7e8c38bce